src/services/execute.ts

```diff
@@ -1,20 +1,30 @@
-import { configuration, loggerService, relay } from '..';
+// SPDX-License-Identifier: Apache-2.0
+import { configuration, loggerService, transport } from '..';
 import apm from '../apm';
 import FRMSMessage from '@tazama-lf/frms-coe-lib/lib/helpers/protobuf';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any -- need FRMS Message object
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-argument -- need FRMS Message object
 export const execute = async (reqObj: any): Promise<void> => {
   let apmTransaction = null;
   try {
-    apmTransaction = apm.startTransaction(`relay-${configuration.DESTINATION_TYPE}`, {
+    loggerService.log('Executing FRMS Relay Service', 'execute');
+    apmTransaction = apm.startTransaction(`relay-${configuration.DESTINATION_TRANSPORT_TYPE}`, {
       childOf: reqObj.metaData?.traceParent ?? undefined,
     });
-    loggerService.log(`relaying to ${configuration.DESTINATION_TYPE}`, 'relay');
-    const message = FRMSMessage.create(reqObj as object);
-    const messageBuffer = FRMSMessage.encode(message).finish();
-    const span = apm.startSpan('relay');
-    await relay.relay(messageBuffer);
-    span?.end();
+    if (configuration.JSON_PAYLOAD === 'false') {
+      const span = apm.startSpan('relay');
+      const msgObj = FRMSMessage.create(reqObj as object);
+      const msgEncoded = FRMSMessage.encode(msgObj).finish();
+
+      await transport.relay(msgEncoded as Buffer);
+      span?.end();
+    } else {
+      const span = apm.startSpan('relay');
+      const messageBuffer = Buffer.from(JSON.stringify(reqObj));
```

Comment:
Why are we creating a `Buffer` from the JSON anyways? I am trying to understand the need of the `data` parameter [here](https://github.com/tazama-lf/frms-coe-lib/blob/e2ef41e77f7cbdf1708bf9fee676f5868c9420b0/src/interfaces/relay-service/ITransportPlugin.ts#L24) (which can be a string):

Both branches effectively relay a buffer through